src/components/data-processing/DataIntegration/FastMNNOptions.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Form, InputNumber, Select, Tooltip, Typography } from 'antd';
+
+import {
+  QuestionCircleOutlined,
+} from '@ant-design/icons';
+
+const { Option } = Select;
+const { Text } = Typography;
+
+const FastMNNOptions = (props) => {
+  const {
+    config, onUpdate, onChange, disabled,
+  } = props;
+
+  const [numGenes, setNumGenes] = useState(config.numGenes);
+
+  return (
+    <>
+      <Form.Item label='# of HGV genes'>
+        <InputNumber
+          value={numGenes}
+          step={100}
+          min={1}
+          onChange={(value) => {
+            onChange();
+            setNumGenes(value);
+          }}
+          onPressEnter={(e) => e.preventDefault()}
+          onStep={(value) => onUpdate({
+            dataIntegration: {
+              methodSettings: {
+                fastmnn: {
+                  numGenes: value,
+                },
+              },
+            },
+          })}
+          onBlur={(e) => onUpdate({
+            dataIntegration: {
+              methodSettings: {
+                fastmnn: {
+                  numGenes: parseInt(e.target.value, 0),
+                },
+              },
+            },
+          })}
+          disabled={disabled}
+        />
+        {' '}
+        <Tooltip overlay={(
+          <span>
+            Number of genes to mark as top highly variable genes (HGV).
+            Integration as well as PCA is based on a sensible selection of HGV.
+            Here, this number selects the top variable genes based on the "vst" method.
+            The default 2000 has been found to be a sensible for many cases.
+            Further info can be found
+            <a
+              href='https://satijalab.org/seurat/articles/integration_introduction.html'
+              target='_blank'
+              rel='noreferrer'
+            >
+              {' '}
+              <code>here</code>
+            </a>
+          </span>
+        )}
+        >
+          <QuestionCircleOutlined />
+        </Tooltip>
+      </Form.Item>
+      <Form.Item label={(
+        <span>
+          Normalization&nbsp;
+          <Tooltip overlay={(
+            <span>
+              Normalization aims to remove technical factors including sequencing depth.
+              There are several methods to achive normalization.
+              "sctransform" claims to recover sharper biological distinction compared to log-normalization.
+              Normalization is applied to each sample before integration.
+              Further info can be found
```

Comment:
Also, it is worth to point out that I personally have no idea what `technical factors` mean. That can just as well be because of my lack of biology/bioinformatics background. I am noting it, because if biologists might not know this, we will need to rephrase it. @vickymorrison any thoughts?

Reply:
Thanks for the review @ivababukova! For this PR I looked at what was already in place in SeuratV4Options to create the counterpart to FastMNN and Unisample. In the new commit Martin has done a great job of refactoring. Regarding `technical errors`, I guess it refers to possible biases that occur when sequencing or in the lab in the sample preparation.

Reply:
For the first sentence, how about: "Normalization aims to remove technical variation that is not biologically relevant, e.g. sequencing depth." ?

Reply:
I like it

Reply:
The thing is that we could add as covariates for the normalization the mt-content, so I guess we should not especify biologically relevant. For the current version we cannot add covariates but I think that we will support it.

Reply:
I would argue that 'not biologically relevant' still stands - if the user chooses to exclude specific gene categories that is because they believe those genes are not biologically relevant to the question they are trying to answer.

Reply:
It makes sense! I'll change!